lib/payrix/resource/merchants.rb

```diff
@@ -1,6 +1,7 @@
 module Payrix
   module Resource
     class Merchants < Base
+      include Mixin::AttrCasters
```

Comment:
It seems Merchants status by default returns an integer and we may not need to include the mixin here. Is there a scenario where it returned a string, as I thought the issue was only with Transactions?

Reply:
Yes thanks for asking about that. I did notice the issue as well with my API testing of merchants, so I added it here for the safety on anyone using the gem, and because we use the merchants endpoint in our use case.

Reply:
To be clear, I was actually seeing this discrepancy for merchants in their webhook alerts.

Here is a redacted alert payload for a merchant in the "not ready" status.

```json
{
  "response":{
    "alert":{
      "subject":"Merchant has been created",
      "entityName":"...",
      "merchantId":"t1_mer_...",
      "merchantDba":null,
      "merchantCreated":"06-10-2022",
      "merchantStatus":"Not Ready",
      "merchantMcc":"",
      "ownerFirstName":"...",
      "ownerLastName":"..."
    },
    "data":[
      {
        "id":"t1_mer_...",
        "created":"2022-06-10 11:22:58.2193",
        "modified":"2022-06-10 11:22:58.2193",
        "creator":"t1_log_...",
        "modifier":"t1_log_...",
        "lastActivity":null,
        "entity":"t1_ent_...",
        "dba":null,
        "new":0,
        "established":"20220609",
        "annualCCSales":0,
        "avgTicket":0,
        "amex":null,
        "discover":null,
        "mcc":null,
        "status":0,
        "boarded":null,
        "inactive":0,
        "frozen":0,
        "environment":"ecommerce",
        "visaMvv":null,
        "chargebackNotificationEmail":null,
        "statusReason":null,
        "totalApprovedSales":0,
        "autoBoarded":null,
        "saqType":null,
        "saqDate":null,
        "qsa":null,
        "letterStatus":0,
        "letterDate":null,
        "tcAttestation":0,
        "visaDisclosure":0,
        "disclosureIP":null,
        "disclosureDate":null,
        "accountClosureReasonCode":null,
        "accountClosureReasonDate":null,
        "annualCCSaleVolume":null,
        "annualACHSaleVolume":null,
        "riskLevel":null,
        "creditRatio":null,
        "creditTimeliness":null,
        "chargebackRatio":null,
        "ndxDays":null,
        "ndxPercentage":null,
        "advancedBilling":null,
        "locationType":null,
        "percentKeyed":null,
        "totalVolume":null,
        "percentEcomm":null,
        "seasonal":null,
        "amexVolume":null,
        "incrementalAuthSupported":null,
        "tmxSessionId":null,
        "percentBusiness":null,
        "applePayActive":"0",
        "applePayStatus":null,
        "googlePayActive":"1"
      }
    ]
  }
}
```

Here is a redacted alert payload for a merchant in the "approved" status.

```json
{
  "response":{
    "alert":{
      "subject":"Merchant has been boarded",
      "entityName":"...",
      "merchantId":"t1_mer_...",
      "merchantDba":null,
      "merchantCreated":"06-07-2022",
      "merchantStatus":"...",
      "merchantMcc":"...",
      "ownerFirstName":"...",
      "ownerLastName":"..."
    },
    "data":[
      {
        "id":"t1_mer_...",
        "created":"2022-06-07 12:56:42.3929",
        "modified":"2022-06-07 12:56:45.3114",
        "creator":"t1_log_...",
        "modifier":"t1_log_...",
        "lastActivity":null,
        "entity":"t1_ent_...",
        "dba":null,
        "new":0,
        "established":20220606,
        "annualCCSales":0,
        "avgTicket":0,
        "amex":null,
        "discover":null,
        "mcc":"8351",
        "status":"2",
        "boarded":"20220607",
        "inactive":0,
        "frozen":0,
        "environment":"cardPresent",
        "visaMvv":null,
        "chargebackNotificationEmail":null,
        "statusReason":null,
        "totalApprovedSales":0,
        "autoBoarded":"1",
        "saqType":null,
        "saqDate":null,
        "qsa":null,
        "letterStatus":0,
        "letterDate":null,
        "tcAttestation":0,
        "visaDisclosure":0,
        "disclosureIP":null,
        "disclosureDate":null,
        "accountClosureReasonCode":null,
        "accountClosureReasonDate":null,
        "annualCCSaleVolume":null,
        "annualACHSaleVolume":null,
        "riskLevel":null,
        "creditRatio":null,
        "creditTimeliness":null,
        "chargebackRatio":null,
        "ndxDays":null,
        "ndxPercentage":null,
        "advancedBilling":null,
        "locationType":null,
        "percentKeyed":null,
        "totalVolume":null,
        "percentEcomm":null,
        "seasonal":null,
        "amexVolume":null,
        "incrementalAuthSupported":null,
        "tmxSessionId":null,
        "percentBusiness":null,
        "applePayActive":1,
        "applePayStatus":null,
        "googlePayActive":1
      }
    ]
  }
}
```

Here you can see `merchants.status` is an integer in the first case and a string in the second case. Funny enough, when I queried the MID's directly, the type was integer for both. So while I can't reproduce the same issue when querying the API for merchants, this creates further uncertainty in my mind, and would make me emphasize/prioritize doing this purely for the safety of it.